pvmismatch/pvmismatch_lib/pvmodule.py

```diff
@@ -235,21 +236,54 @@ def setSuns(self, Ee, cells=None):
         """
         if cells is None:
             if np.isscalar(Ee):
-                for pvc in self.pvcells:
+                new_pvcells = range(self.numberCells)  # new list of cells
+                old_pvcells = dict.fromkeys(self.pvcells)  # same as set(pvcells)
+                for cell_id, pvcell in enumerate(self.pvcells):
+                    if old_pvcells[pvcell] is None:
+                        new_pvcells[cell_id] = copy(pvcell)
+                        old_pvcells[pvcell] = new_pvcells[cell_id]
+                    else:
+                        new_pvcells[cell_id] = old_pvcells[pvcell]
+                self.pvcells = new_pvcells
+                for pvc in set(self.pvcells):
```

Comment:
Good point, but I think we should still change it, in favor of a something that's already been evaluated, If the name is the issue, we can rename first to make it clearer:

    pvcell_set = old_pvcells.itervalues()  # iterator over new copies of unique cells
    for pvc in pvcell_set:
        do stuff ...